py/desitarget/lyazcat.py

```diff
@@ -0,0 +1,441 @@
+    
+    Parameters
+    ----------
+    tile : :class:'str'
+        The TILEID of the tile to process.
+    night : :class:'str'
+        The date associated with the observation of the 'tile' used.
+        * Must be in YYYYMMDD format
+    petal_num : :class:'int'
+        If 'all_petals' isn't used, the single petal to create a zcat for.
+    zcatdir : :class:'str'
+        The location for the daily redrock output.
+    outputdir : :class:'str'
+        The filepath to write out the zcat file.
+    qn_flag : :class:'bool'
+        Flag to add QuasarNP data (or not) to the zcat file.
+    sq_flag : :class:'bool'
+        Flag to add SQUEzE data (or not) to the zcat file.
+    abs_flag : :class:'bool'
+        Flag to add MgII Absorption data (or not) to the zcat file.
+    squeze_model : :class:'str' or 'numpy.array'
+        The filename and path for the SQUEzE model file. IF this function
+        is run as part of a loop, the calling function will load the model
+        file and this will be an array instead.
+        
+    Outputs
+    -------
+    A FITS catalog that incorporates redrock, QuasarNP, SQUEzE, and MgII
+    absorption redshifts and confidence values.
+    
+    Notes
+    -----
+    - There will be many here.
+    """
+    # EBL Load the SQUEzE Model file first. This is a very large file,
+    # so if multiple petals are to be processed, we only want to load
+    # it into memory once.
+    if isinstance(squeze_model, str) and sq_flag:
+        tmark('    Loading SQUEzE Model file')
+        squeze_model = Model.from_json(load_json(squeze_model))
+        tmark('      Model file loaded')
+    
+    # EBL Create the filepath for the input tile/night combination
+    rootdir = os.path.join(zcatdir, 'tiles', 'cumulative')
+    tiledir = os.path.join(rootdir, tile)
+    ymdir = os.path.join(tiledir, night)
+        
+    # EBL Create the filename tag that appends to zbest-*, coadd-*, and zqso-*
+    # files.
+    filename_tag = f'{petal_num}-{tile}-thru{night}.fits'
+    zbestname = f'zbest-{filename_tag}'
+    coaddname = f'coadd-{filename_tag}'
+    outputname = f'zqso-{filename_tag}'
+    
+    zcat = make_new_zcat(os.path.join(ymdir, zbestname))
+    if isinstance(zcat, bool):
+        print('  !!!! Petal Number does not have a corresponding zbest file !!!!')
+    else:
+        if qn_flag :
+            zcat = add_qn_data(zcat, os.path.join(ymdir, coaddname))
+        if sq_flag:
+            zcat = add_sq_data(zcat, os.path.join(ymdir, coaddname), squeze_model)
+        if abs_flag:
+            zcat = add_abs_data(zcat, os.path.join(ymdir, coaddname))
+        
+        fzcat = zcomb_selector(zcat)
+    
+        full_outputname = zcat_writer(outputdir,fzcat,outputname, qn_flag,
+                                      sq_flag, abs_flag)
+    
+        tmark('--{} written out correctly.'.format(full_outputname))
+        print('===================================')
+
+    
+if __name__=='__main__':
```

Comment:
The formalism in `desitarget` is to have separate scripts in `bin` instead of using `if __name__=='__main__'` in the module itself. See the many examples in [the bin directory](https://github.com/desihub/desitarget/tree/master/bin).

Reply:
I haven't done it that way before, so I'll look at that tonight to make the changeover when I add the MgII absorption functionality.

Reply:
@geordie666 I switched this over to using the bin script method.